gobblin-utility/src/main/java/gobblin/util/SchedulerUtils.java

```diff
@@ -226,18 +236,15 @@ private static void loadJobConfigsRecursive(List<Properties> jobConfigs, Propert
   }
 
   private static Set<String> getJobConfigurationFileExtensions(Properties properties) {
-    Iterable<String> jobConfigFileExtensionsIterable = Splitter.on(",").omitEmptyStrings().trimResults().split(
-        properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_EXTENSIONS_KEY,
+    Iterable<String> jobConfigFileExtensionsIterable = Splitter.on(",").omitEmptyStrings().trimResults()
+        .split(properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_EXTENSIONS_KEY,
             ConfigurationKeys.DEFAULT_JOB_CONFIG_FILE_EXTENSIONS));
-    return Sets.newHashSet(Iterables.transform(jobConfigFileExtensionsIterable, new Function<String, String>() {
-          @Override
-          public String apply(String input) {
-            if (input == null) {
-              throw new NullPointerException("Given input is null: " + input);
-            }
-            return input.toLowerCase();
-          }
-        }));
+    return ImmutableSet.copyOf(Iterables.transform(jobConfigFileExtensionsIterable, new Function<String, String>() {
```

Comment:
Yeah, we could use a case insensitive `TreeSet`, but to me that is more for sorting.